### DISCUSSION TITLE
Button: In the state without styles, the enumeration type of "size" is insufficient for the theme I need.

### DISCUSSION THREAD

**Discussion by zhanghengxin** [primefaces/community#584] [PrimeVue]
![image](https://github.com/primefaces/community/assets/24197448/a4254578-1eb0-41c1-82b5-18f7b39d3fa6)

![image](https://github.com/primefaces/community/assets/24197448/1d79fe3a-c18a-4c03-a5ab-05045f3e411c)

Approach to the solution>> Size, custom property.

**Comment by zhanghengxin** [comment]
@melloware @cagataycivici @atakantepe 

If you want to create a style-less mode, this type of functionality needs to be well-supported: Custom types have limited my theme. Custom themes require support for more types.

**Comment by tugcekucukoglu** [comment]
Button has 3 different sizes. The size property, whose default is null, has two different supports: 'small' and 'large'. You can customize this further with pt property or create your own custom button.

**Reply by zhanghengxin** [reply]
但是 ts 会因为类型原因报错，所以，我的需求是，类型应该可以被扩展。


However, TypeScript will throw errors due to type constraints. Therefore, my requirement is that types should be extensible.
